crates/bevy_ecs/src/query/fetch.rs

```diff
@@ -1513,7 +1569,8 @@ macro_rules! impl_anytuple_fetch {
                 $(
                     $name.1 = $name::matches_component_set($state, &|id| _table.has_column(id));
                     if $name.1 {
-                        $name::set_table(&mut $name.0, $state, _table);
+                         // SAFETY: The invariants are uphold by the caller.
```

Comment:
```suggestion
                         // SAFETY: The invariants are required to be upheld by the caller.
```

This is also sort of incomplete as a safety comment, but given this PR is focused on the lint, I won't block on this.